optuna/multi_objective/samplers/_motpe.py

```diff
@@ -0,0 +1,639 @@
+            :class:`~optuna.distributions.LogUniformDistribution`,
+            :class:`~optuna.distributions.IntUniformDistribution`,
+            or :class:`~optuna.distributions.IntLogUniformDistribution`.
+        prior_weight:
+            The weight of the prior. This argument is used in
+            :class:`~optuna.distributions.UniformDistribution`,
+            :class:`~optuna.distributions.DiscreteUniformDistribution`,
+            :class:`~optuna.distributions.LogUniformDistribution`,
+            :class:`~optuna.distributions.IntUniformDistribution`,
+            :class:`~optuna.distributions.IntLogUniformDistribution`, and
+            :class:`~optuna.distributions.CategoricalDistribution`.
+        consider_magic_clip:
+            Enable a heuristic to limit the smallest variances of Gaussians used in
+            the Parzen estimator.
+        consider_endpoints:
+            Take endpoints of domains into account when calculating variances of Gaussians
+            in Parzen estimator. See the original paper for details on the heuristics
+            to calculate the variances.
+        n_startup_trials:
+            The random sampling is used instead of the MOTPE algorithm until the given number
+            of trials finish in the same study. 11 * number of variables - 1 is recommended in the
+            original paper.
+        n_ehvi_candidates:
+            Number of candidate samples used to calculate the expected hypervolume improvement.
+        gamma:
+            A function that takes the number of finished trials and returns the number of trials to
+            form a density function for samples with low grains. See the original paper for more
+            details.
+        weights:
+            A function that takes the number of finished trials and returns a weight for them. As
+            default, weights are automatically calculated by the MOTPE's default strategy.
+        seed:
+            Seed for random number generator.
+
+    .. note::
+        Initialization with Latin hypercube sampling may improve optimization performance.
+        However, the current implementation only supports initialization with random sampling.
+
+    Example:
+
+        .. testcode::
+
+                import numpy as np
+
+                seed = 128
+                num_variables = 9
+                n_startup_trials = 11 * num_variables - 1
+
+                def objective(trial):
+                    x = []
+                    for i in range(1, num_variables + 1):
+                        x.append(trial.suggest_uniform(f"x{i}", 0.0, 2.0 * i))
+                    return x
+
+                sampler = optuna.multi_objective.samplers.MOTPEMultiObjectiveSampler(
+                    n_startup_trials=n_startup_trials,
+                    n_ehvi_candidates=24,
+                    seed=seed
+                )
+                study.optimize(objective, n_trials=250)
```

Comment:
IMO, this example is not executable since `optuna` is not imported and `study` is undefined. I think this is trivial issue and I can update this example in a follow-up PR.

(The example actually passed the CI, but IMO the `doctest` should be able to detect these errors.)